### PR TITLE
Add meal category from api ts draft2

### PR DESCRIPTION
**Pull request by tauri-st**
Looping into recipes data to access key value pairs in the tags.

The comments within the code explain each step, and trying out each console.log one at a time shows you which data I get at each step.

1. allRecipeTags shows that each recipe has an array of tags
each-recipe-array-of-tags

2. The first for...in loop shows all the tags as individual objects
each-tag-object

3. Making them into an array, which got kinda funky and showed each tag object had been put alone into it's own array
tag-object-all-key-values-in-one-array

4. For...of loop with a nested for...in loop at ln 68 and 69 shows each individual key value pair in all of the tag objects
all-tag-object-individual-key-value-pairs

I had a lot of problems pushing updates to the last draft. It said the tip of my branch was not up to date and I needed to pull. I had already made sure it was up to date with development (and made sure again twice), then it was still recommending I pull so I tried to pull from the branch already pushed as a draft but it said it does not appear to be a repo. I saved everything in screenshots, and was really tired by now so I just made a new branch off of the old one, named it draft2 and pushed that. Sorry for the messiness y'all

### CODE REVIEW

File: docs/js/main.js [hunk 1]
```diff
@@ -56,6 +55,22 @@ const showRecipes = function (recipes) {
     for (const recipe in recipes) {
         const recipeID = recipes[recipe].id;
         const title = recipes[recipe].name;
+        //show array of tags for each recipe
+        const allRecipeTags = recipes[recipe].tags;
+        //console.log(allRecipeTags);
+        //show each tag object
+        for (let tag in allRecipeTags) {
+            //console.log(tag, allRecipeTags[tag]);
+            let allTags = (tag, allRecipeTags[tag]);
```
Comment (terrifricker) [inline docs/js/main.js:64]: I don't understand this line of code.

File: docs/js/main.js [hunk 2]
```diff
@@ -56,6 +55,22 @@ const showRecipes = function (recipes) {
     for (const recipe in recipes) {
         const recipeID = recipes[recipe].id;
         const title = recipes[recipe].name;
+        //show array of tags for each recipe
+        const allRecipeTags = recipes[recipe].tags;
+        //console.log(allRecipeTags);
+        //show each tag object
+        for (let tag in allRecipeTags) {
```
Comment (terrifricker) [inline docs/js/main.js:62]: ~~This is a loop for objects, but allRecipeTags is an array.~~

File: docs/js/main.js (hunk 1 above)
Comment (terrifricker) [inline docs/js/main.js:64]: I'm not sure what this line means...

File: docs/js/main.js [hunk 3]
```diff
@@ -56,6 +55,22 @@ const showRecipes = function (recipes) {
...
+            //console.log(tag, allRecipeTags[tag]);
+            let allTags = (tag, allRecipeTags[tag]);
+            let tagsArray = [];
```
Comment (terrifricker) [inline docs/js/main.js:65]: You already have an array of all the tag objects in variable allRecipeTags

File: docs/js/main.js [hunk 4]
```diff
@@ -56,6 +55,22 @@ const showRecipes = function (recipes) {
...
+            tagsArray.push(allTags);
+            //console.log(tagsArray);
+            for (let keys of tagsArray) {
+                for (let values in keys) {
+                    console.log(values, keys[values]);
```
Comment (terrifricker) [inline docs/js/main.js:70]: Is this still just for figuring out what you are looking for?

File: docs/js/main.js (hunk 2 above)
Comment (tauri-st) [reply docs/js/main.js:62]: That makes sense, not the right loop

File: docs/js/main.js (hunk 3 above)
Comment (tauri-st) [reply docs/js/main.js:65]: The last for...of, for...in loop is what allows you to access the key:value pairs in each tag object, and consoling it out shows each of them individually so I THOUGHT I was getting somewhere and now had the loop to work within in order to mess with a conditional with an includes method or something

File: docs/js/main.js (hunk 2 above)
Comment (terrifricker) [reply docs/js/main.js:62]: You do need for .. in, I guess.  I think because it is an array of objects.

File: docs/js/main.js (hunk 1 above)
Comment (tauri-st) [reply docs/js/main.js:64]: I definitely think I got into overkill land here. So I was taught that in order to loop through multiple objects and access their key value pairs, you would need to put all of the objects in an array. So I overkilled it by trying to console out each tag THEN force it into an array (no successfully), which required a loop. I ALREADY have an array of the tags in allRecipe tags x.x

**Comment by terrifricker** [comment]
Closing because it was replaced by pull request #82

**Pull request closed without merging**